lib/src/logging_service.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:meta/meta.dart';
+
+/// Exception thrown on logging service configuration errors.
+class LoggingException extends Error implements Exception {
+  /// Message describing the exception.
+  final String message;
+
+  LoggingException(this.message);
+
+  @override
+  String toString() => 'Logging exception: $message';
+}
+
+/// Manages logging services.
+///
+/// The [LoggingService] is intended for internal use only.
+///
+/// * To create logging channels, see [LoggingChannel].
+/// * To enable or disable a logging channel, use [debugEnableLogging].
+/// * To query channel enablement, use [debugShouldLogEvent].
+class LoggingService {
+  /// The shared service instance.
+  static final LoggingService instance = LoggingService();
```

Comment:
Instead of a static singleton, what about a top-level singleton `LoggingService`? It could be defined in `lib/logs.dart` or in `lib/src/logs.dart`.

Reply:
Sure!

Reply:
I kept it in `logging_service` for now since I don't think we want to export it.  We can certainly change that though (or `hide` it on export).